src/components/LoginRegister/Login.js

```diff
@@ -17,6 +17,7 @@ export default function Login() {
     if(login) history.push("/app");
   }, [login, history])
 
+  // comment on extraction
```

Comment:
so i wrote up in the new utility file that you can "see below" for why i wrote it. A couple things are going on here that I really don't like:

1. Your component is **tightly** coupled to the network request. There is almost no way we could mock this out from a testing perspective unless our tests "knew" everything about the network request.
2. this function gets redefined every single time this component renders. It's a small thing, but it doesn't need to be redefined every time the component renders.
3. The logic of the network request is potentially reusable. with it living in the component here, there is no way another component could use it.
4. the component becomes harder to read and track what's going on when the asynchronous code is right in line. Ideally, your component can treat the imported network request code as a "black box" and not have to worry about the details of the request itself. Just needs to tell the functon/hook what to do if it succeeds or fails.